tests/webTables.spec.ts

```diff
@@ -21,38 +21,27 @@ test.describe("Web Tables-Owners", async () => {
   test("Validate search by Last Name", async ({ page }) => {
     const lastNameInputfield = page.getByRole("textbox");
     const findOwnerButton = page.getByRole("button", { name: "Find Owner" });
-
-    await lastNameInputfield.click();
-    await lastNameInputfield.fill("Black");
-    await findOwnerButton.click();
-
-    await expect(lastNameInputfield).toHaveValue("Black");
-
-    await lastNameInputfield.click();
-    await lastNameInputfield.clear();
-    await lastNameInputfield.fill("Davis");
-    await findOwnerButton.click();
-
-    const ownerFullNameRows = page.locator("tbody tr");
-
-    for (let row of await ownerFullNameRows.all()) {
-      await expect(row.locator("td").first()).toContainText("Davis");
-    }
-
-    await lastNameInputfield.click();
-    await lastNameInputfield.clear();
-    await lastNameInputfield.fill("Es");
-    await findOwnerButton.click();
-
-    for (let row of await ownerFullNameRows.all()) {
-      await expect(row.locator("td").first()).toContainText("Es");
-    }
-
-    await lastNameInputfield.click();
-    await lastNameInputfield.clear();
-    await lastNameInputfield.fill("Playwright");
-    await findOwnerButton.click();
-    expect(page.getByText('No owners with LastName starting with "Playwright"'));
+    const ownerLastNames = ["Black", "Davis", "Es"];
+
+    for(let ownerLastName of ownerLastNames){
+
+      await lastNameInputfield.clear();
+      await lastNameInputfield.fill(ownerLastName);
+      await findOwnerButton.click();
+      await page.waitForTimeout(1000);
+
+      const ownerFullNameRows = page.locator("tbody tr");
+
+    for (const row of await ownerFullNameRows.all()) {
+      const fullName = await row.locator("td").first().textContent();
+      
+      if (ownerLastName == "Playwright") {
+          await expect(page.getByText('No owners with LastName starting with "Playwright"')).toBeVisible();
+      } else {
+          expect(fullName).toContain(ownerLastName);
+      }
+  }
```

Comment:
Okay, you are very close.
The array in line 24 is missing the value "Playwright", so line 38 is never executed as TURE because `ownerLastName` variable never has an assigned value "Playwright".

Hardcoded wait in line 31, replace with a dynamic wait. Don't just delete it :) Replace is with "waitForResponse()" and find in the networking tab the API which is triggered whtn you click Find Owner button, so your script will wait until that API call will be completed.

Instead of using Generic Assertion in line 41, you can use a locator assertion, so line 36 can be be deleted.
Line 39 should be replaced with different find of assertion. Instead of validating the locator visibility, need to replace it with assertion that validates text displayed in the table.